### PR TITLE
Add custom exception for the HTTP 451 response

### PR DESCRIPTION
**Pull request by devkhan**
Implements #1237.

As per this [blog](https://developer.github.com/changes/#--the-451-status-code-is-now-supported) post.

GitHub API will now return a HTTP 451 response (rather than a 403) when a repository or gist cannot be accessed due to a DMCA notice.

Checklist:

- [x] Add new exception.
- [x] Add it to `HandleErrors`.
- [x] `.\build FixProjects`
- [x] Unit tests.
- [x] Integration tests.

/cc @ryangribble

### CODE REVIEW

**Comment by shiftkey** [comment]
@devkhan just added a task to ensure the files are synced across all the projects

**Comment by devkhan** [comment]
@ryangribble The Travis Linux build is failing. Can you please review?

File: Octokit.Tests/Exceptions/LegalRestrictionExceptionTests.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using Xunit;
+
+namespace Octokit.Tests.Exceptions
+{
+    public class LegalRestrictionExceptionTests
+    {
+        [Fact]
+        public void HasDefaultMessage()
+        {
+            var response = new Response(HttpStatusCode.Forbidden, null, new Dictionary<string, string>(), "application/json");
```
Comment (ryangribble) [inline Octokit.Tests/Exceptions/LegalRestrictionExceptionTests.cs]: Shouldnt the status code here be the HTTP 451 (not Forbidden?)

Comment (devkhan) [inline Octokit.Tests/Exceptions/LegalRestrictionExceptionTests.cs]: I'm really sorry for this, I'll fix this. Thanks.

Comment (ryangribble) [inline Octokit.Tests/Exceptions/LegalRestrictionExceptionTests.cs]: No need to apologosie!  Everyone's contributions are much appreciated  :grinning:

**Comment by shiftkey** [comment]
LGTM :thumbsup:

**Comment by devkhan** [comment]
Should I add some integration tests using one of the repos from <http://github.com/github/dmca> ?

**Comment by shiftkey** [comment]
@devkhan don't worry about it. I think this area of the codebase is rather boring and unit tests are good enough coverage for the error code handling...

**Pull request merged**